### PR TITLE
📝💼 adding template files for magento

### PR DESCRIPTION
**Pull request by filmaj**
These changes are in prep for integrating Magento open source approval process into Adobe's internal one. Would like to have the template files that we want Magento projects to generally use handy in this repo as well.

- contributing guidelines shamelessly copied from magento/magento2
- OSL license
- copying.txt from the magento2 repo
- added example magento copyright header to license header doc.

Especially interested in @dmanners' take on this to make sure I have this right 😄

### CODE REVIEW

**Comment by dmanners** [comment]
It looks good to in the sense that the files are the ones from our repo but it does remind me that I should update these files to match our new beginners guides from https://github.com/magento/magento2/wiki/Getting-Started

**Comment by filmaj** [comment]
@dmanners OK, well, I can update the template files in here to reflect whatever changes end up landing. Let me know!

**Comment by dmanners** [comment]
I have added a new PR with my initial thoughts and further questions to our devdocs writer and head of open source https://github.com/magento/magento2/pull/22951 will let you know when we get some update on this.

File: LICENSE_HEADER.md [hunk 1]
```diff
@@ -58,6 +58,15 @@ rem OF ANY KIND, either express or implied. See the License for the specific lan
 rem governing permissions and limitations under the License.
 ```
 
+# Magento Projects
+
+### Text
+
+```
+Copyright © Magento, Inc. All rights reserved.
```
Comment (stevengill) [inline LICENSE_HEADER.md]: Should the copyright be updated to Adobe?

Comment (macdonst) [reply LICENSE_HEADER.md]: I think Magento is being retained as a separate copyright.

Comment (filmaj) [reply LICENSE_HEADER.md]: Yes, chatted with @dmanners about this in the opensourceoffice Slack and we decided to stay off this for now.

File: .github/CONTRIBUTING_MAGENTO.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+1. Search current [listed issues](https://github.com/magento/magento2/issues) (open or closed) for similar proposals of intended contribution before starting work on a new contribution.
+2. Review the [Contributor License Agreement](https://magento.com/legaldocuments/mca) if this is your first time contributing.
+3. Create and test your work.
+4. Fork the Magento 2 repository according to the [Fork A Repository instructions](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#fork) and when you are ready to send us a pull request – follow the [Create A Pull Request instructions](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#pull_request).
+5. Once your contribution is received the Magento 2 development team will review the contribution and collaborate with you as needed.
+
+## Code of Conduct
+
+Please note that this project is released with a Contributor Code of Conduct. We expect you to agree to its terms when participating in this project.
+The full text is available in the repository [Wiki](https://github.com/magento/magento2/wiki/Magento-Code-of-Conduct).
```
[the remark this answers is not in the document]
Comment (filmaj) [reply .github/CONTRIBUTING_MAGENTO.md]: I looked at it, it's based on the contributor covenant one as well. When stars align!

File: COPYING.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Copyright © 2013-present Magento, Inc.
```
[the remark this answers is not in the document]
Comment (filmaj) [reply COPYING.txt]: IIRC the end date doesn't make a difference for copyright notices. All that matters is the start date.

**Pull request merged**